workflow/snakemake_rules/nextclade.smk

```diff
@@ -103,7 +103,6 @@ rule run_nextclade:
     params:
         nextclade_input_dir = temp(directory(f"data/{database}/nextclade_inputs")),
         nextclade_output_dir = temp(directory(f"data/{database}/nextclade")),
-    threads: 64
```

Comment:
We usually run ingest with 16 CPUs, assigning 64 threads means that this rule cannot run in parallel to other rules which slows down ingest.

Since we use incremental alignments, the alignment part is fast, so we can just let Nextclade use all cores available to get it done with - this may be slightly subideal from a performance point of view but fine-tuning threads for different parallel rules is a nightmare and not worth it.

For the foreseeable future no problem will arise from parallelizing this rule.

Reply:
Making sure I understand the effect and rationale here: by removing the `threads:` parameter and it's passing into `./bin/run-nextclade` (where it ends up as `nextclade run --jobs N`), Nextclade will automatically use all cores available (as it did before when available cores were <= 64) but Snakemake's scheduler will no longer know that. Instead, Snakemake will consider this `run_nextclade` rule to only consume 1 core/thread and thus schedule other jobs to run concurrently.

[Looking at the DAG](https://tsibley.net/paste/2022-11-03wTFK2IDq-ncov-ingest-dag.svg), it seems like the `download_previous_alignment` rule and a bunch of `upload_single` rule jobs and the `upload_raw_ndjson` rule are all candidates for concurrent execution with `run_nextclade` now. These call `download-from-s3` or `upload-from-s3`, which also use all cores available and also aren't annotated to Snakemake as such, so decompression/compression jobs will be in CPU contention with the Nextclade alignment.

The rationale is that this isn't much of an issue in practice and the Nextclade alignment job is fast anyway (at least when not contending for CPU?) and the conceptual overhead of accounting for concurrency is too hard/not worth it.

Have I got that all right? :-)

Reply:
Yes exactly @tsibley, well put!

The only way this oversubscription of threads could harm us is if the bottleneck process that runs on say 1 thread doesn't run at 100% as a result.

If this was the case, we may want to limit threads for some of the compression jobs, but not for Nextclade as it's so fast in most situations anyways.

Reply:
Got it. Thanks for confirming!